searchcore/src/vespa/searchcore/proton/matching/match_tools.cpp

```diff
@@ -214,4 +219,52 @@ std::unique_ptr<IDiversifier> MatchToolsFactory::createDiversifier() const
                                    _diversityParams.cutoff_strategy == DiversityParams::CutoffStrategy::STRICT);
 }
 
+std::unique_ptr<AttributeOperationTask>
+MatchToolsFactory::createTask(vespalib::stringref attribute, vespalib::stringref operation) const {
+    return (!attribute.empty() && ! operation.empty())
+           ? std::make_unique<AttributeOperationTask>(_requestContext, attribute, operation)
+           : std::unique_ptr<AttributeOperationTask>();
+}
+std::unique_ptr<AttributeOperationTask>
+MatchToolsFactory::createOnMatchTask() const {
+    return createTask(execute::onmatch::Attribute::lookup(_queryEnv.getProperties()),
```

Comment:
are we sure we want people to set this in the query and not in the config?

Reply:
No, nothing definitive here. But as you need to specify 'mutable' you will not accidentally run into it. WE will wait until we see use cases more clearly.